File: src/ProjectR.Scripting/content/scripts/spells/SummonFieryInfernals.lua
function GetTargetType()
    return Myself
end

function GetName()
    return "Summon: Steam Duo"
end

function GetDescription()
    return "Summons I.C. and Fiahr to overwhelm your enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 2
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    local minion2 = SummonMinionCopy(a, "Fiahr")
    minion2:ClearSpells()
    minion2:AddSpell("Netherball")
    minion2:AddSpell("Netherball")
    minion2:AddSpell("Extinction")
    minion2:AddSpell("Hellfire Flare")
    minion2:AddSpell("Hell's Inferno")
    minion2:AddSpell("Nuclear Meltdown")
    minion2:AddSpell("Ragnarok")

    local minion = SummonMinionCopy(a, "I.C.")
    minion:ClearSpells()
    minion:AddSpell("Netherball")
    minion:AddSpell("Netherball")
    minion:AddSpell("Frost Blast")
    minion:AddSpell("Frozen Blood")
    minion:AddSpell("Ice Prison")
    minion:AddSpell("Icicle Barrage")
    minion:AddSpell("Breath of the Baby Ice Dragon")
end

File: src/ProjectR.Scripting/content/scripts/spells/SummonFieryInfernals.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class SummonFieryInfernals : SpellScriptBase
    {
        public override string Name { get { return "Summon: Steam Duo"; } }
        public override string Description { get { return "Summons I.C. and Fiahr to overwhelm your enemies."; } }

        public override TargetType TargetType { get { return TargetType.Myself; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 200; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster)
        {
            var minion = SummonMinionCopy(caster, "Fiahr");
            minion.Spells.Clear();
            AddSpell(minion, "Netherball");
            AddSpell(minion, "Netherball");
            AddSpell(minion, "Extinction");
            AddSpell(minion, "Hellfire Flare");
            AddSpell(minion, "Hell's Inferno");
            AddSpell(minion, "Nuclear Meltdown");
            AddSpell(minion, "Ragnarok");

            var minion2 = SummonMinionCopy(caster, "I.C.");
            minion2.Spells.Clear();
            AddSpell(minion2, "Netherball");
            AddSpell(minion2, "Netherball");
            AddSpell(minion2, "Frost Blast");
            AddSpell(minion2, "Frozen Blood");
            AddSpell(minion2, "Ice Prison");
            AddSpell(minion2, "Icicle Barrage");
            AddSpell(minion2, "Breath of the Baby Ice Dragon");
        }
    }
}